File: HackerRank/maxsubarray.py
t = input()
for i in range(t):
	index = 0
	n = input()
	k = raw_input()
	a = k.split(' ')
	large = int(a[0])
	for j in range(n):
		a[j] = int(a[j])
		if a[j]>large:
			index = j
	
	noncontig = a[index]
	contig = a[index]
	for k in range(n):
		if a[k]>0 and k!=index:
			noncontig+=a[k]

	for k in range(0,n-1):
		total = a[k]
		for p in range(k+1,n):
			total+=a[p]
			if total>contig:
				contig = total

	print contig,
	print noncontig

File: HackerRank/maxsubarray.c
#include<stdio.h>
void main(){
	int large, a[100000], t, n, index=0, contig=0, noncontig=0, i, j, k, total=0, p;
	scanf("%d",&t);
	for(i=0;i<t;i++){
		scanf("%d",&n);
		for(j=0;j<n;j++){
			scanf("%d",&a[j]);
			if(j==0)
				large = a[0];
			if(a[j]>large)
				index = j;
		}
		noncontig = a[index];
		contig = a[index];

		for(k=0;k<n;k++){
			if(a[k]>0 && k!=index)
				noncontig+=a[k];
		}

		for(k=0;k<n-1;k++){
			total = a[k];
			for(p=k+1;p<n;p++){
				total+=a[p];
				if(total>contig)
					contig = total;
			}
		}
		printf("%d %d",contig,noncontig);	
	}
}